scene/animation/animation_player.cpp

```diff
@@ -579,6 +580,10 @@ bool AnimationPlayer::is_playing() const {
 	return playing;
 }
 
+bool AnimationPlayer::is_paused() const {
+	return paused;
+}
+
```

Comment:
```suggestion
bool AnimationPlayer::is_paused() const {
	return !playing && playback.current.from;
}
```
I assume above code should work. The fundamental issue should be that `playback.current.from` isn't exposed to GDScript. Alternatively, you might consider simply exposing `AnimationPlayer::is_valid()` without implementing is_paused.

At this point, the animation will be paused after the animation duration has elapsed, but internally it is correct to determine that state is paused.

If you want to distinguish that the animation has stopped after its duration has elapsed, it should not be difficult for external code to compare the current time with the animation duration.

So I say again, the only thing missing here is the exposure of `playback.current.from`.

Reply:
Sorry, I have mistook `playback.current.from` for a variable storing the current position in the animation 😅.

Yes I did expose is_valid and the behavior seems to work flawlessly. I'll Update the PR accordingly.

Reply:
Sorry. I mistook `playback.current.from` for the current position in the animation. 😅. I went on on test exposing `is_valid()`, and it worked flawlessly! I will go on about updating the PR accordingly.